server/src/main/java/org/elasticsearch/rest/action/cat/RestNodesAction.java

```diff
@@ -530,6 +524,14 @@ Table buildTable(
         return table;
     }
 
+    public static String getRoleAbbreviationString(DiscoveryNode node) {
+        if (node.getRoles().isEmpty()) {
+            return "-";
+        } else {
+            return node.getRoles().stream().map(DiscoveryNodeRole::roleNameAbbreviation).sorted().collect(Collectors.joining());
+        }
+    }
```

Comment:
Could it be useful outside rest actions? May be it is worth moving to `DiscoveryNode`?

Reply:
Good point, we do almost the same thing for `DiscoveryNode#toString`.